### PR TITLE
Set compile warning as error for core projects

### PR DESCRIPTION
**Pull request by yiylin**
#### What does this implement/fix? Briefly explain your changes.
1. Set compile warning as error for core projects
2.  Fix warning related errors

### CODE REVIEW

**Review by harsha-simhadri — APPROVED** [review]

File: src/index.cpp [hunk 1]
```diff
@@ -3020,12 +3021,12 @@ template <typename T, typename TagT, typename LabelT> void Index<T, TagT, LabelT
     for (uint32_t i = 0; i < _nd; i++)
     {
         char *cur_node_offset = _opt_graph + i * _node_size;
-        float cur_norm = dist_fast->norm(_data + i * _aligned_dim, _aligned_dim);
+        float cur_norm = dist_fast->norm(_data + i * _aligned_dim, (unsigned int)_aligned_dim);
```
Comment (gopalrs) [inline src/index.cpp:3024]: Should be uint32_t

File: src/pq_flash_index.cpp [hunk 2]
```diff
@@ -1292,8 +1292,8 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
             compute_dists(node_nbrs, nnbrs, dist_scratch);
             if (stats != nullptr)
             {
-                stats->n_cmps += (double)nnbrs;
-                stats->cpu_us += (double)cpu_timer.elapsed();
+                stats->n_cmps += (unsigned int)nnbrs;
```
Comment (gopalrs) [inline src/pq_flash_index.cpp:1295]: Should be uint32_t

File: src/pq_flash_index.cpp [hunk 3]
```diff
@@ -1359,8 +1359,8 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
...
```
Comment (gopalrs) [inline src/pq_flash_index.cpp:1362]: uint32_t

**Review by gopalrs — CHANGES_REQUESTED** [review]

File: src/index.cpp (hunk 1 above)
Comment (yiylin) [reply src/index.cpp]: changed

File: src/pq_flash_index.cpp (hunk 3 above)
Comment (yiylin) [reply src/pq_flash_index.cpp]: changed

File: src/pq_flash_index.cpp (hunk 2 above)
Comment (yiylin) [reply src/pq_flash_index.cpp]: changed

**Pull request merged**